src/utils.ts

```diff
@@ -26,7 +26,10 @@ export default class Utils {
     /** Function that can be used to wait for a condition before returning. */
     async wait<T>(condition: () => T | false, timeout = 5000, check = 100): Promise<T> {
         return await new Promise((resolve, reject) => {
-            setTimeout(() => reject("TIMEOUT"), timeout);
+            setTimeout(() => {
+                clearInterval(interval);
```

Comment:
How is this compiling. Shouldn't this be moved to after defining the interval?

Reply:
I can move it if you wish but the reason this is valid is because the use of interval here occurs in a closure. Closures can access variables that are defined after the fact.
![image](https://user-images.githubusercontent.com/16382065/147764651-80129fa7-abc2-4814-9558-272dff805ae6.png)
As demonstrated here